src/app/components/challenge/challengeleaderboard/challengeleaderboard.component.ts

```diff
@@ -256,7 +273,7 @@ export class ChallengeleaderboardComponent implements OnInit, AfterViewInit {
    * @param self  context value of this
    */
   updateLeaderboardResults(leaderboardApi, self) {
-    const leaderboard = leaderboardApi.slice();
+    const leaderboard = (Array.isArray(leaderboardApi) ? leaderboardApi.slice() : [leaderboardApi]);
```

Comment:
Why is this change required here?

Reply:
Both `fetchLeaderboard()` and `refreshLeaderboard()` calls `updateLeaderboardResults()`, yet the argument `refreshLeaderboard()` gives differs from what `fetchLeaderboard()` gives. The solution to this difference is to wrap an object in array if it's not already an array. Let me know if you want to change this.

Reply:
changed!

Reply:
But they return the same response as they call the same API I think.

Reply:
Okay, I shall change it then. I don't think it would work though.

Reply:
yes please :+1: